pipeline/inspection.py

```diff
@@ -40,6 +40,20 @@ def convert_value(cls, v):
         elif isinstance(v, (str)):
             return extract_first_number(v)
         return None
+    
+class GuaranteedAnalysis(BaseModel):
+    title: Optional[str] = None
+    nutrients: List[NutrientValue] = []
+
+    @model_validator(mode='before')
```

Comment:
Why not use the `field` or `fields` parameter of model_validator?

Reply:
You mean `field_validator`? I don't see a field parameter in the [pydantic doc](https://docs.pydantic.dev/latest/api/functional_validators/#model-validators).

I do this because the LLM sometimes just returns null on those fields but the pydantic expects a list.

Reply:
Sorry, I was confusing model_validator with [field_validator](https://docs.pydantic.dev/latest/api/functional_validators/#pydantic.functional_validators.field_validator). Please check if it could simplify the validation. Otherwise, this works.

Reply:
It's because the field_validator runs after the model_validator. If I use the field_validator the model will fail.

Reply:
We were able to resolve the issue and use field_validator.